lib/docker2aci.go

```diff
@@ -124,31 +124,25 @@ func convertReal(backend internal.Docker2ACIBackend, dockerURL string, squash bo
 
 	conversionStore := newConversionStore()
 
-	var images acirenderer.Images
-	var aciLayerPaths []string
-	var curPwl []string
-	for i := len(ancestry) - 1; i >= 0; i-- {
-		layerID := ancestry[i]
-
-		// only compress individual layers if we're not squashing
-		layerCompression := compression
-		if squash {
-			layerCompression = common.NoCompression
-		}
+	// only compress individual layers if we're not squashing
+	layerCompression := compression
+	if squash {
+		layerCompression = common.NoCompression
+	}
 
-		aciPath, manifest, err := backend.BuildACI(i, layerID, parsedDockerURL, layersOutputDir, tmpDir, curPwl, layerCompression)
-		if err != nil {
-			return nil, fmt.Errorf("error building layer: %v", err)
-		}
+	aciLayerPaths, aciManifests, err := backend.BuildACI(ancestry, parsedDockerURL, layersOutputDir, tmpDir, layerCompression)
+	if err != nil {
+		return nil, err
+	}
 
-		key, err := conversionStore.WriteACI(aciPath)
+	var images acirenderer.Images
+	for i, aciLayerPath := range aciLayerPaths {
```

Comment:
Although `aciLayerPath` has the right order (from layer `len(ancestry) - 1` to 0) the index `i` is ascending in this loop.

You need to change the `uint16(i)` below to `uint16(len(aciLayerPaths) - 1 - i)`.